vendor/bat-native-ads/src/bat/ads/internal/account/confirmations/confirmations_state.cc

```diff
@@ -676,19 +676,52 @@ bool ConfirmationsState::ParseAdRewardsFromDictionary(
     return false;
   }
 
-  const base::Value* unblinded_tokens =
-      dictionary->FindListKey("unblinded_payment_tokens");
-  if (unblinded_tokens && unblinded_tokens->GetList().empty()) {
-    // Migration path for https://github.com/brave/brave-browser/issues/16678
-    ad_rewards_dictionary->SetDoubleKey(
-        "unreconciled_estimated_pending_rewards", 0.0);
-  }
+  // Migration path for https://github.com/brave/brave-browser/issues/16678 and
+  // https://github.com/brave/brave-browser/issues/16744
+  const double value = GetUnreconciledEstimatedPendingRewards(dictionary);
+  ad_rewards_dictionary->SetDoubleKey("unreconciled_estimated_pending_rewards",
+                                      value);
 
   ad_rewards_->SetFromDictionary(ad_rewards_dictionary);
 
   return true;
 }
 
+double ConfirmationsState::GetUnreconciledEstimatedPendingRewards(
+    base::DictionaryValue* dictionary) {
+  DCHECK(dictionary);
+
+  const base::Value* unblinded_payment_tokens =
+      dictionary->FindListKey("unblinded_payment_tokens");
+  if (!unblinded_payment_tokens) {
+    return 0.0;
+  }
+
+  size_t count = unblinded_payment_tokens->GetList().size();
+  if (count == 0) {
+    // There are no uncleared unblinded payment tokens to redeem
+    return {};
```

Comment:
Any reason to use braces here rather than `0.0` as above?

Reply:
Fixed